cli/central-client/src/main/java/org/ballerinalang/central/client/Utils.java

```diff
@@ -129,9 +130,23 @@ public static void createBalaInHomeRepo(Response balaDownloadResponse, Path pkgP
                     logFormatter.formatLog("error accessing bala : " + balaCacheWithPkgPath.toString()));
         }
 
-        createBalaFileDirectory(balaCacheWithPkgPath, logFormatter);
-        writeBalaFile(balaDownloadResponse, balaCacheWithPkgPath.resolve(balaFile), pkgOrg + "/" + pkgName +
+        // Create the following temp path bala/<org-name>/<pkg-name>/<pkg-version>/temp
+        Path tempPath = pkgPathInBalaCache.resolve(validPkgVersion).resolve("temp");
+        createBalaFileDirectory(tempPath, logFormatter);
+
+        // Write balaFiles to tempPath
+        writeBalaFile(balaDownloadResponse, tempPath.resolve(balaFile),
```

Comment:
What happens when there is a temp location is left over from the previous run? Does it override?

Reply:
Since we are renaming the `temp` file, if it runs successfully there wont be any leftovers. However, if there is an issue, and a temp file remains, it will be overwritten with the newly downloaded deps and it will attempt to rename the temp folder again